src/lib/comms/sol-coap.c

```diff
@@ -104,6 +105,7 @@ struct resource_observer {
 };
 
 struct pending_reply {
+    struct sol_reentrant reentrant;
```

Comment:
I don't think I can move it to the last place, because the ```pending_reply``` is a variable length structure. I'll move it after ```bool observing```. OK?